Build failure: cie-middleware-linux

## Steps To Reproduce

Steps to reproduce the behavior:

1. build *cie-middleware-linux*

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
unpacking source archive /nix/store/m0v20ix69f04ygla0h0cp8h39d14xicx-source
source root is source
Running phase: patchPhase
applying patch /nix/store/iwfjnihf24bjnmqcpys6msdla8ykxjr4-use-system-podofo.patch
patching file libs/meson.build
patching file libs/sign-sdk/include/PdfSignatureGenerator.h
patching file libs/sign-sdk/src/PdfSignatureGenerator.cpp
patching file libs/sign-sdk/src/disigonsdk.cpp
substituteStream() in derivation cie-middleware-linux-1.5.2: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'libs/pkcs11/src/CSP/AbilitaCIE.cpp')
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
/build/source/libs /build/source
/build/tmp.aEBuoyPdnG /build/source/libs /build/source
/build/source/libs /build/source
Owner: CN=example.org, OU=a, O=a, L=a, ST=a, C=AL
Issuer: CN=example.org, OU=a, O=a, L=a, ST=a, C=AL
Serial number: 6b7d8d2bccbceab912b7de0a6b98166edc550e50
Valid from: Thu Jan 09 18:01:30 UTC 2025 until: Fri Jan 10 18:01:30 UTC 2025
Certificate fingerprints:
         SHA1: 55:D1:A3:8A:9C:6E:A5:33:E3:7C:8C:BB:F2:19:DB:43:86:E4:A7:3C
         SHA256: 63:4F:5F:2B:8B:3B:7C:B0:D9:8A:5C:B6:EF:43:35:D1:E5:4E:39:84:01:A8:05:2B:35:F5:DA:4A:37:8F:A2:5A
Signature algorithm name: SHA256withRSA
Subject Public Key Algorithm: 2048-bit RSA key
Version: 3

Extensions: 

#1: ObjectId: 2.5.29.35 Criticality=false
AuthorityKeyIdentifier [
KeyIdentifier [
0000: BC BB 2C 24 10 DD B5 17   2F 3E 2E E6 7F 7D 34 64  ..,$..../>....4d
0010: 0E F5 DE CB                                        ....
]
]

#2: ObjectId: 2.5.29.19 Criticality=true
BasicConstraints:[
  CA:true
  PathLen: no limit
]

#3: ObjectId: 2.5.29.14 Criticality=false
SubjectKeyIdentifier [
KeyIdentifier [
0000: BC BB 2C 24 10 DD B5 17   2F 3E 2E E6 7F 7D 34 64  ..,$..../>....4d
0010: 0E F5 DE CB                                        ....
]
]

Trust this certificate? [no]:  Certificate was added to keystore
mesonConfigurePhase flags: --prefix=/nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2 --libdir=/nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/lib --libexecdir=/nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/libexec --bindir=/nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/bin --sbindir=/nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/sbin --includedir=/nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/include --mandir=/nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/share/man --infodir=/nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/share/info --localedir=/nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/share/locale -Dauto_features=enabled -Dwrap_mode=nodownload --buildtype=plain
The Meson build system
Version: 1.6.0
Source dir: /build/source/libs
Build dir: /build/source/libs/build
Build type: native build
Project name: libcie-pkcs11
Project version: 1.5.2
C compiler for the host machine: gcc (gcc 14.2.1 "gcc (GCC) 14.2.1 20241116")
C linker for the host machine: gcc ld.bfd 2.43.1
C++ compiler for the host machine: g++ (gcc 14.2.1 "g++ (GCC) 14.2.1 20241116")
C++ linker for the host machine: g++ ld.bfd 2.43.1
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: YES (/nix/store/0d4m43yp69lrm8imxbqgl9zxjwwz52jw-pkg-config-wrapper-0.29.2/bin/pkg-config) 0.29.2
Run-time dependency libcrypto found: YES 3.3.2
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency libcrypto++ found: NO (tried pkgconfig and cmake)
Run-time dependency libcryptopp found: YES 8.9
Run-time dependency cryptopp found: NO (tried pkgconfig and cmake)
Run-time dependency libpcsclite found: YES 2.3.0
Run-time dependency threads found: YES
Run-time dependency libcurl found: YES 8.11.0
Run-time dependency fontconfig found: YES 2.15.0
Run-time dependency freetype2 found: YES 26.2.20
Run-time dependency libpng found: YES 1.6.43
Run-time dependency libpodofo found: YES
Run-time dependency libxml-2.0 found: YES 2.13.5
Run-time dependency xml2 found: NO (tried pkgconfig and cmake)
Run-time dependency zlib found: YES 1.3.1
Build targets in project: 2

libcie-pkcs11 1.5.2

  User defined options
    auto_features: enabled
    bindir       : /nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/bin
    buildtype    : plain
    includedir   : /nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/include
    infodir      : /nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/share/info
    libdir       : /nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/lib
    libexecdir   : /nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/libexec
    localedir    : /nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/share/locale
    mandir       : /nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/share/man
    prefix       : /nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2
    sbindir      : /nix/store/m2j2k633dqlfvmxh3jrf9v0jdi33y7xa-cie-middleware-linux-1.5.2/sbin
    wrap_mode    : nodownload

Found ninja-1.12.1 at /nix/store/byvgvqlh007flwk940c99zfgzrjg0zsg-ninja-1.12.1/bin/ninja
Running phase: buildPhase
build flags: -j1
[1/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_disigonsdk.cpp.o
../sign-sdk/src/disigonsdk.cpp: In function 'long int disigon_set_int(int, int)':
../sign-sdk/src/disigonsdk.cpp:136:30: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  136 |   return disigon_set(option, (void*)value);
      |                              ^~~~~~~~~~~~
../sign-sdk/src/disigonsdk.cpp: In function 'long int disigon_sign_set_int(DISIGON_CTX, int, int)':
../sign-sdk/src/disigonsdk.cpp:188:40: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  188 |   return disigon_sign_set(ctx, option, (void*)value);
      |                                        ^~~~~~~~~~~~
../sign-sdk/src/disigonsdk.cpp: In function 'long int disigon_verify_set_int(DISIGON_CTX, int, int)':
../sign-sdk/src/disigonsdk.cpp:551:42: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  551 |   return disigon_verify_set(ctx, option, (void*)value);
      |                                          ^~~~~~~~~~~~
In file included from ../shared/src/Sign/disigonsdk.h:16,
                 from ../sign-sdk/src/disigonsdk.cpp:4:
../sign-sdk/src/disigonsdk.cpp: In function 'long int disigon_verify_cleanup_result(VERIFY_RESULT*)':
../sign-sdk/src/disigonsdk.cpp:762:27: warning: deleting 'void*' is undefined [-Wdelete-incomplete]
  761 |           SAFEDELETE(pVerifyResult->verifyInfo.pSignerInfos->pSignerInfo[i]
      |                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  762 |                          .pTimeStamp);
      |                          ~^~~~~~~~~~
../shared/src/Sign/definitions.h:121:14: note: in definition of macro 'SAFEDELETE'
  121 |       delete pointer;       \
      |              ^~~~~~~
../sign-sdk/src/disigonsdk.cpp: In function 'long int sign_xml(DISIGON_SIGN_CONTEXT*, UUCByteArray&)':
../sign-sdk/src/disigonsdk.cpp:1595:42: warning: '.xml' directive writing 4 bytes into a region of size between 1 and 1024 [-Wformat-overflow=]
 1595 |       sprintf(pContext->szOutputFile, "%s.xml", pContext->szInputFile);
      |                                          ^~~~
In file included from /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/stdio.h:970,
                 from /nix/store/7hwyxbsi39mzrl4kmqs0m478wd2pg63x-libxml2-2.13.5-dev/include/libxml2/libxml/tree.h:24,
                 from /nix/store/7hwyxbsi39mzrl4kmqs0m478wd2pg63x-libxml2-2.13.5-dev/include/libxml2/libxml/parser.h:16,
                 from /nix/store/7hwyxbsi39mzrl4kmqs0m478wd2pg63x-libxml2-2.13.5-dev/include/libxml2/libxml/tree.h:17,
                 from ../sign-sdk/src/disigonsdk.cpp:6:
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'long int sign_xml(DISIGON_SIGN_CONTEXT*, UUCByteArray&)' at ../sign-sdk/src/disigonsdk.cpp:1595:14:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output between 5 and 1028 bytes into a destination of size 1024
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
../sign-sdk/src/disigonsdk.cpp: In function 'long int sign_xml(DISIGON_SIGN_CONTEXT*, UUCByteArray&)':
../sign-sdk/src/disigonsdk.cpp:1593:47: warning: '%s' directive writing up to 1023 bytes into a region of size 1017 [-Wformat-overflow=]
 1593 |       sprintf(pContext->szOutputFile, "signed_%s.xml", pContext->szInputFile);
      |                                               ^~
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'long int sign_xml(DISIGON_SIGN_CONTEXT*, UUCByteArray&)' at ../sign-sdk/src/disigonsdk.cpp:1593:14:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output between 12 and 1035 bytes into a destination of size 1024
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
../sign-sdk/src/disigonsdk.cpp: In function 'long int sign_pdf(DISIGON_SIGN_CONTEXT*, UUCByteArray&)':
../sign-sdk/src/disigonsdk.cpp:1710:40: warning: '.pdf' directive writing 4 bytes into a region of size between 1 and 1024 [-Wformat-overflow=]
 1710 |     sprintf(pContext->szOutputFile, "%s.pdf", pContext->szInputFile);
      |                                        ^~~~
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'long int sign_pdf(DISIGON_SIGN_CONTEXT*, UUCByteArray&)' at ../sign-sdk/src/disigonsdk.cpp:1710:12:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output between 5 and 1028 bytes into a destination of size 1024
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
../sign-sdk/src/disigonsdk.cpp: In function 'long int disigon_sign_sign(DISIGON_CTX)':
../sign-sdk/src/disigonsdk.cpp:481:40: warning: '.p7m' directive writing 4 bytes into a region of size between 1 and 1024 [-Wformat-overflow=]
  481 |     sprintf(pContext->szOutputFile, "%s.p7m", pContext->szInputFile);
      |                                        ^~~~
In function 'int sprintf(char*, const char*, ...)',
    inlined from 'long int disigon_sign_sign(DISIGON_CTX)' at ../sign-sdk/src/disigonsdk.cpp:481:12:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/stdio2.h:30:34: note: '__builtin___sprintf_chk' output between 5 and 1028 bytes into a destination of size 1024
   30 |   return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   31 |                                   __glibc_objsize (__s), __fmt,
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   32 |                                   __va_arg_pack ());
      |                                   ~~~~~~~~~~~~~~~~~
[2/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_definitions.cpp.o
[3/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_XAdESVerifier.cpp.o
[4/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_XAdESGenerator.cpp.o
[5/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_Util_util.cpp.o
../sign-sdk/src/Util/util.cpp: In function 'void PutPaddingBT0(ByteArray&, long int)':
../sign-sdk/src/Util/util.cpp:166:17: warning: comparison of integer expressions of different signedness: 'long int' and 'size_t' {aka 'long unsigned int'} [-Wsign-compare]
  166 |       if (dwLen > ba.size()) throw logged_error("Lunghezza del padding errata");
      |           ~~~~~~^~~~~~~~~~~
[6/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_UUCTextFileWriter.cpp.o
[7/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_UUCTextFileReader.cpp.o
[8/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_UUCStringTable.cpp.o
[9/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_UUCProperties.cpp.o
[10/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_UUCLogger.cpp.o
../sign-sdk/src/UUCLogger.cpp: In member function 'void UUCLogger::log(unsigned int, const char*, unsigned int, const char*)':
../sign-sdk/src/UUCLogger.cpp:34:13: warning: comparison of integer expressions of different signedness: 'const unsigned int' and 'int' [-Wsign-compare]
   34 |   if (nType > m_nLogLevel) return;
      |       ~~~~~~^~~~~~~~~~~~~
[11/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_TSAClient.cpp.o
[12/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_SignerInfoGenerator.cpp.o
[13/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_SignedDocument.cpp.o
../sign-sdk/src/SignedDocument.cpp: In constructor 'CSignedDocument::CSignedDocument(const BYTE*, int)':
../sign-sdk/src/SignedDocument.cpp:56:48: warning: comparison of integer expressions of different signedness: 'size_t' {aka 'long unsigned int'} and 'int' [-Wsign-compare]
   56 |         if (strlen(szEncoded) + strlen(szPart) > len) throw -1;
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
../sign-sdk/src/SignedDocument.cpp: In member function 'int CSignedDocument::verify(const char*)':
../sign-sdk/src/SignedDocument.cpp:131:21: warning: comparison of integer expressions of different signedness: 'int' and 'unsigned int' [-Wsign-compare]
  131 |   for (int i = 0; i < m_signerInfos.size(); i++) {
      |                   ~~^~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/SignedDocument.cpp: In member function 'CCertificate CSignedDocument::getSignerCertificate(int)':
../sign-sdk/src/SignedDocument.cpp:181:21: warning: comparison of integer expressions of different signedness: 'int' and 'unsigned int' [-Wsign-compare]
  181 |   for (int i = 0; i < m_certificates.size(); i++) {
      |                   ~~^~~~~~~~~~~~~~~~~~~~~~~
[14/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_SignedDataGeneratorEx.cpp.o
In file included from ../sign-sdk/src/ASN1/ASN1Object.h:4,
                 from ../sign-sdk/src/ASN1/ASN1GenericSequence.h:4,
                 from ../sign-sdk/src/ASN1/ASN1Setof.h:4,
                 from ../sign-sdk/include/SignedDataGeneratorEx.h:13,
                 from ../sign-sdk/src/SignedDataGeneratorEx.cpp:10:
../sign-sdk/src/SignedDataGeneratorEx.cpp: In member function 'void SignedDataGeneratorEx::toByteArray(UUCByteArray&)':
../shared/src/Sign/definitions.h:73:19: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
   73 | #define szDataOID "1.2.840.113549.1.7.1"
      |                   ^~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/SignedDataGeneratorEx.cpp:133:30: note: in expansion of macro 'szDataOID'
  133 |     CContentType contentType(szDataOID);
      |                              ^~~~~~~~~
[15/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_SignatureGenerator.cpp.o
../sign-sdk/src/SignatureGenerator.cpp: In member function 'virtual long int CSignatureGenerator::Generate(UUCByteArray&, BOOL, BOOL)':
../sign-sdk/src/SignatureGenerator.cpp:168:17: warning: comparison of integer expressions of different signedness: 'int' and 'unsigned int' [-Wsign-compare]
  168 |   for (i = 0; i < m_digestAlgos.size(); i++) {
      |               ~~^~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/SignatureGenerator.cpp:172:9: warning: comparison of integer expressions of different signedness: 'int' and 'unsigned int' [-Wsign-compare]
  172 |   if (i == m_digestAlgos.size()) m_digestAlgos.addElement(hashOID);
      |       ~~^~~~~~~~~~~~~~~~~~~~~~~
In file included from ../sign-sdk/src/ASN1/ASN1Object.h:4,
                 from ../sign-sdk/src/ASN1/ASN1GenericSequence.h:4,
                 from ../sign-sdk/src/ASN1/ASN1Setof.h:4,
                 from ../sign-sdk/src/ASN1/SignerInfo.h:4,
                 from ../sign-sdk/include/SignatureGenerator.h:3,
                 from ../sign-sdk/src/SignatureGenerator.cpp:2:
../shared/src/Sign/definitions.h:73:19: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
   73 | #define szDataOID "1.2.840.113549.1.7.1"
      |                   ^~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/SignatureGenerator.cpp:324:30: note: in expansion of macro 'szDataOID'
  324 |     CContentType contentType(szDataOID);
      |                              ^~~~~~~~~
../sign-sdk/src/SignatureGenerator.cpp:247:48: warning: 'hash' may be used uninitialized [-Wmaybe-uninitialized]
  247 |     CASN1OctetString digestString(hash, hashlen);
      |                                                ^
../sign-sdk/src/SignatureGenerator.cpp:177:9: note: 'hash' was declared here
  177 |   BYTE* hash;
      |         ^~~~
../sign-sdk/src/SignatureGenerator.cpp:247:48: warning: 'hashlen' may be used uninitialized [-Wmaybe-uninitialized]
  247 |     CASN1OctetString digestString(hash, hashlen);
      |                                                ^
../sign-sdk/src/SignatureGenerator.cpp:178:7: note: 'hashlen' was declared here
  178 |   int hashlen;
      |       ^~~~~~~
[16/153] Compiling C object libcie-sign-sdk.a.p/sign-sdk_src_RSA_sha2.c.o
[17/153] Compiling C object libcie-sign-sdk.a.p/sign-sdk_src_RSA_sha1.c.o
[18/153] Compiling C object libcie-sign-sdk.a.p/sign-sdk_src_RSA_rsa.c.o
[19/153] Compiling C object libcie-sign-sdk.a.p/sign-sdk_src_RSA_rc2.c.o
[20/153] Compiling C object libcie-sign-sdk.a.p/sign-sdk_src_RSA_r_stdlib.c.o
[21/153] Compiling C object libcie-sign-sdk.a.p/sign-sdk_src_RSA_r_encode.c.o
../sign-sdk/src/RSA/r_encode.c:176:35: warning: argument 1 of type 'unsigned char *' declared as a pointer [-Warray-parameter=]
  176 | static void encode(unsigned char *f, unsigned char *p, int count) {
      |                    ~~~~~~~~~~~~~~~^
In file included from ../sign-sdk/src/RSA/rsaeuro.h:45,
                 from ../sign-sdk/src/RSA/r_encode.c:33:
../sign-sdk/src/RSA/r_encode.c:93:32: note: previously declared as an array 'unsigned char[4]'
   93 | static void encode PROTO_LIST((unsigned char[4], unsigned char[3], int));
      |                                ^~~~~~~~~~~~~~~~
../sign-sdk/src/RSA/global.h:63:26: note: in definition of macro 'PROTO_LIST'
   63 | #define PROTO_LIST(list) list
      |                          ^~~~
../sign-sdk/src/RSA/r_encode.c:176:53: warning: argument 2 of type 'unsigned char *' declared as a pointer [-Warray-parameter=]
  176 | static void encode(unsigned char *f, unsigned char *p, int count) {
      |                                      ~~~~~~~~~~~~~~~^
../sign-sdk/src/RSA/r_encode.c:93:50: note: previously declared as an array 'unsigned char[3]'
   93 | static void encode PROTO_LIST((unsigned char[4], unsigned char[3], int));
      |                                                  ^~~~~~~~~~~~~~~~
../sign-sdk/src/RSA/global.h:63:26: note: in definition of macro 'PROTO_LIST'
   63 | #define PROTO_LIST(list) list
      |                          ^~~~
[22/153] Compiling C object libcie-sign-sdk.a.p/sign-sdk_src_RSA_nn.c.o
[23/153] Compiling C object libcie-sign-sdk.a.p/sign-sdk_src_RSA_desc.c.o
../sign-sdk/src/RSA/desc.c:447:19: warning: argument 1 of type 'UINT4[32]' {aka 'long unsigned int[32]'} with mismatched bound [-Warray-parameter=]
  447 | void deskey(UINT4 subkeys[32], unsigned char key[8], int encrypt) {
      |             ~~~~~~^~~~~~~~~~~
In file included from ../sign-sdk/src/RSA/desc.c:40:
../sign-sdk/src/RSA/desc.c:175:25: note: previously declared as 'UINT4 *' {aka 'long unsigned int *'}
  175 | void deskey PROTO_LIST((UINT4 *, unsigned char *, int));
      |                         ^~~~~~~
../sign-sdk/src/RSA/des.h:60:26: note: in definition of macro 'PROTO_LIST'
   60 | #define PROTO_LIST(list) list
      |                          ^~~~
../sign-sdk/src/RSA/desc.c:447:46: warning: argument 2 of type 'unsigned char[8]' with mismatched bound [-Warray-parameter=]
  447 | void deskey(UINT4 subkeys[32], unsigned char key[8], int encrypt) {
      |                                ~~~~~~~~~~~~~~^~~~~~
../sign-sdk/src/RSA/desc.c:175:34: note: previously declared as 'unsigned char *'
  175 | void deskey PROTO_LIST((UINT4 *, unsigned char *, int));
      |                                  ^~~~~~~~~~~~~~~
../sign-sdk/src/RSA/des.h:60:26: note: in definition of macro 'PROTO_LIST'
   60 | #define PROTO_LIST(list) list
      |                          ^~~~
[24/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_PdfVerifier.cpp.o
../sign-sdk/src/PdfVerifier.cpp: In member function 'int PDFVerifier::VerifySignature(int, const char*, char*, REVOCATION_INFO*)':
../sign-sdk/src/PdfVerifier.cpp:230:13: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<const PoDoFo::PdfObject*>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
  230 |   if (index >= signatureVector.size()) return 0;
      |       ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/PdfVerifier.cpp: In member function 'int PDFVerifier::GetSignature(int, UUCByteArray&, SignatureAppearanceInfo&)':
../sign-sdk/src/PdfVerifier.cpp:392:13: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<const PoDoFo::PdfObject*>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
  392 |   if (index >= signatureVector.size()) return -8;
      |       ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
[25/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_PdfSignatureGenerator.cpp.o
../sign-sdk/src/PdfSignatureGenerator.cpp: In member function 'void PdfSignatureGenerator::InitSignature(int, float, float, float, float, const char*, const char*, const char*, const char*, const char*, const char*, const char*, const char*, const char*, const char*, const char*, const char*)':
../sign-sdk/src/PdfSignatureGenerator.cpp:213:33: warning: catching polymorphic type 'class PoDoFo::PdfError' by value [-Wcatch-value=]
  213 |     } catch (::PoDoFo::PdfError err) {
      |                                 ^~~
[26/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_M7MParser.cpp.o
[27/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_CounterSignatureGenerator.cpp.o
../sign-sdk/src/CounterSignatureGenerator.cpp: In constructor 'CounterSignatureGenerator::CounterSignatureGenerator(CSignedDocument&, int)':
../sign-sdk/src/CounterSignatureGenerator.cpp:15:46: warning: member 'CounterSignatureGenerator::m_signerInfoIndex' is used uninitialized [-Wuninitialized]
   15 |       m_signerInfo(m_signedDoc.getSignerInfo(m_signerInfoIndex)),
      |                                              ^~~~~~~~~~~~~~~~~
../sign-sdk/src/CounterSignatureGenerator.cpp: In member function 'void CounterSignatureGenerator::toByteArray(UUCByteArray&)':
../sign-sdk/src/CounterSignatureGenerator.cpp:82:17: warning: comparison of integer expressions of different signedness: 'int' and 'unsigned int' [-Wsign-compare]
   82 |   for (i = 0; i < m_certificates.size(); i++) {
      |               ~~^~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CounterSignatureGenerator.cpp:86:9: warning: comparison of integer expressions of different signedness: 'int' and 'unsigned int' [-Wsign-compare]
   86 |   if (i == m_certificates.size()) m_certificates.addElement(cert);
      |       ~~^~~~~~~~~~~~~~~~~~~~~~~~
[28/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_CertStore.cpp.o
[29/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_CSP_IAS.cpp.o
../sign-sdk/src/CSP/IAS.cpp: In member function 'ByteDynArray IAS::SM(ByteArray&, ByteArray&, ByteArray&, ByteArray&)':
../sign-sdk/src/CSP/IAS.cpp:668:39: warning: comparison of integer expressions of different signedness: 'size_t' {aka 'long unsigned int'} and 'int' [-Wsign-compare]
  668 |   if (apdu.size() == 5 || apdu.size() == (apdu[4] + 6)) {
      |                           ~~~~~~~~~~~~^~~~~~~~~~~~~~~~
[30/153] Compiling C++ object libcie-sign-sdk.a.p/sign-sdk_src_CIESigner.cpp.o
../sign-sdk/src/CIESigner.cpp: In member function 'long int CCIESigner::Init(const char*)':
../sign-sdk/src/CIESigner.cpp:55:24: warning: catching polymorphic type 'class scard_error' by value [-Wcatch-value=]
   55 |   } catch (scard_error err) {
      |                        ^~~
../sign-sdk/src/CIESigner.cpp: In member function 'virtual long int CCIESigner::Sign(UUCByteArray&, UUCByteArray&, int, UUCByteArray&)':
../sign-sdk/src/CIESigner.cpp:124:24: warning: catching polymorphic type 'class scard_error' by value [-Wcatch-value=]
  124 |   } catch (scard_error err) {
      |                        ^~~
[31/153] Compiling C object libcie-sign-sdk.a.p/sign-sdk_src_CIEEngineHelper.c.o
[32/153] Compiling C object libcie-sign-sdk.a.p/sign-sdk_src_CIEEngine.c.o
FAILED: libcie-sign-sdk.a.p/sign-sdk_src_CIEEngine.c.o 
gcc -Ilibcie-sign-sdk.a.p -I. -I.. -I../sign-sdk/include -I../sign-sdk/src -I../shared/src -I/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include -I/nix/store/9y9568l8m27khl30y9nn67cxwpfxxcqb-crypto++-8.9.0-dev/include -I/nix/store/7hwyxbsi39mzrl4kmqs0m478wd2pg63x-libxml2-2.13.5-dev/include/libxml2 -I/nix/store/583bivq800mxhip4nrc1glv8xsgc730r-pcsclite-2.3.0-dev/include/PCSC -I/nix/store/g26bs63hz87c9s3sg0v42d6d8gjk36qw-curl-8.11.0-dev/include -I/nix/store/x6wh1a1xnyzvh4k0x6ggdr8w9w38zl52-fontconfig-2.15.0-dev/include -I/nix/store/wc97xp20zf23vx6rj65ah6knqpqwn9bn-freetype-2.13.3-dev/include/freetype2 -I/nix/store/j0cx534qlwqdinajjm4f7l3c2460785l-libpng-apng-1.6.43-dev/include/libpng16 -I/nix/store/b1hp07c057fmvh1j6lmrdirs20r0jsf1-podofo-0.9.8-dev/include/podofo -I/nix/store/1lggwqzapn5mn49l9zy4h566ysv9kzdb-zlib-1.3.1-dev/include -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -fPIC -pthread -MD -MQ libcie-sign-sdk.a.p/sign-sdk_src_CIEEngine.c.o -MF libcie-sign-sdk.a.p/sign-sdk_src_CIEEngine.c.o.d -o libcie-sign-sdk.a.p/sign-sdk_src_CIEEngine.c.o -c ../sign-sdk/src/CIEEngine.c
../sign-sdk/src/CIEEngine.c: In function 'cie_pkey_rsa_sign':
../sign-sdk/src/CIEEngine.c:216:3: warning: 'EVP_PKEY_get0_RSA' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  216 |   rsa = EVP_PKEY_get0_RSA(pkey);
      |   ^~~
In file included from /nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/rand.h:23,
                 from /nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:29,
                 from ../shared/src/Sign/CIEEngine.h:12,
                 from ../sign-sdk/src/CIEEngine.c:8:
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/evp.h:1356:22: note: declared here
 1356 | const struct rsa_st *EVP_PKEY_get0_RSA(const EVP_PKEY *pkey);
      |                      ^~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:216:7: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  216 |   rsa = EVP_PKEY_get0_RSA(pkey);
      |       ^
../sign-sdk/src/CIEEngine.c:254:27: warning: passing argument 2 of 'makeDigestInfo' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  254 |   makeDigestInfo(md_type, tbs, tbslen, digestinfo, &digestinfolen);
      |                           ^~~
../sign-sdk/src/CIEEngine.c:56:46: note: expected 'unsigned char *' but argument is of type 'const unsigned char *'
   56 | int makeDigestInfo(int algid, unsigned char *pbtDigest, size_t btDigestLen,
      |                               ~~~~~~~~~~~~~~~^~~~~~~~~
../sign-sdk/src/CIEEngine.c: In function 'cie_get_ec_key_method':
../sign-sdk/src/CIEEngine.c:316:5: warning: 'EC_KEY_METHOD_new' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  316 |     ops = EC_KEY_METHOD_new((EC_KEY_METHOD *)EC_KEY_OpenSSL());
      |     ^~~
In file included from /nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:28:
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/ec.h:1480:38: note: declared here
 1480 | OSSL_DEPRECATEDIN_3_0 EC_KEY_METHOD *EC_KEY_METHOD_new(const EC_KEY_METHOD *meth);
      |                                      ^~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:316:5: warning: 'EC_KEY_OpenSSL' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  316 |     ops = EC_KEY_METHOD_new((EC_KEY_METHOD *)EC_KEY_OpenSSL());
      |     ^~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/ec.h:1298:44: note: declared here
 1298 | OSSL_DEPRECATEDIN_3_0 const EC_KEY_METHOD *EC_KEY_OpenSSL(void);
      |                                            ^~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:317:5: warning: 'EC_KEY_METHOD_get_sign' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  317 |     EC_KEY_METHOD_get_sign(ops, &orig_sign, NULL, NULL);
      |     ^~~~~~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/ec.h:1543:28: note: declared here
 1543 | OSSL_DEPRECATEDIN_3_0 void EC_KEY_METHOD_get_sign
      |                            ^~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:318:5: warning: 'EC_KEY_METHOD_set_sign' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  318 |     EC_KEY_METHOD_set_sign(ops, orig_sign, NULL, &orig_sign);
      |     ^~~~~~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/ec.h:1499:28: note: declared here
 1499 | OSSL_DEPRECATEDIN_3_0 void EC_KEY_METHOD_set_sign
      |                            ^~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:318:50: error: passing argument 4 of 'EC_KEY_METHOD_set_sign' from incompatible pointer type [-Wincompatible-pointer-types]
  318 |     EC_KEY_METHOD_set_sign(ops, orig_sign, NULL, &orig_sign);
      |                                                  ^~~~~~~~~~
      |                                                  |
      |                                                  int (**)(int,  const unsigned char *, int,  unsigned char *, unsigned int *, const BIGNUM *, const BIGNUM *, EC_KEY *) {aka int (**)(int,  const unsigned char *, int,  unsigned char *, unsigned int *, const struct bignum_st *, const struct bignum_st *, struct ec_key_st *)}
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/ec.h:1508:37: note: expected 'ECDSA_SIG * (*)(const unsigned char *, int,  const BIGNUM *, const BIGNUM *, EC_KEY *)' {aka 'struct ECDSA_SIG_st * (*)(const unsigned char *, int,  const struct bignum_st *, const struct bignum_st *, struct ec_key_st *)'} but argument is of type 'int (**)(int,  const unsigned char *, int,  unsigned char *, unsigned int *, const BIGNUM *, const BIGNUM *, EC_KEY *)' {aka 'int (**)(int,  const unsigned char *, int,  unsigned char *, unsigned int *, const struct bignum_st *, const struct bignum_st *, struct ec_key_st *)'}
 1508 |                        ECDSA_SIG *(*sign_sig)(const unsigned char *dgst,
      |                        ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 1509 |                                               int dgst_len,
      |                                               ~~~~~~~~~~~~~
 1510 |                                               const BIGNUM *in_kinv,
      |                                               ~~~~~~~~~~~~~~~~~~~~~~
 1511 |                                               const BIGNUM *in_r,
      |                                               ~~~~~~~~~~~~~~~~~~~
 1512 |                                               EC_KEY *eckey));
      |                                               ~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:319:5: warning: 'EC_KEY_METHOD_get_compute_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  319 |     EC_KEY_METHOD_get_compute_key(ops, &ossl_ecdh_compute_key);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/ec.h:1536:28: note: declared here
 1536 | OSSL_DEPRECATEDIN_3_0 void EC_KEY_METHOD_get_compute_key
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:320:5: warning: 'EC_KEY_METHOD_set_compute_key' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  320 |     EC_KEY_METHOD_set_compute_key(ops, cie_ecdh_compute_key);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/ec.h:1494:28: note: declared here
 1494 | OSSL_DEPRECATEDIN_3_0 void EC_KEY_METHOD_set_compute_key
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c: In function 'cie_engine_ctrl':
../sign-sdk/src/CIEEngine.c:383:47: error: passing argument 2 of 'd2i_X509' from incompatible pointer type [-Wincompatible-pointer-types]
  383 |         cie_x509_certificate = d2i_X509(NULL, &cie_certificate, cie_certlen);
      |                                               ^~~~~~~~~~~~~~~~
      |                                               |
      |                                               unsigned char **
In file included from /nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/rsa.h:21,
                 from /nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:25:
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/x509.h:753:1: note: expected 'const unsigned char **' but argument is of type 'unsigned char **'
  753 | DECLARE_ASN1_FUNCTIONS(X509)
      | ^~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c: In function 'cie_pkey_method_rsa':
../sign-sdk/src/CIEEngine.c:433:3: warning: 'EVP_PKEY_meth_find' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  433 |   orig_meth = (EVP_PKEY_METHOD *)EVP_PKEY_meth_find(EVP_PKEY_RSA);
      |   ^~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/evp.h:1753:46: note: declared here
 1753 | OSSL_DEPRECATEDIN_3_0 const EVP_PKEY_METHOD *EVP_PKEY_meth_find(int type);
      |                                              ^~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:434:3: warning: 'EVP_PKEY_meth_get_sign' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  434 |   EVP_PKEY_meth_get_sign(orig_meth, &orig_pkey_rsa_sign_init,
      |   ^~~~~~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/evp.h:2092:28: note: declared here
 2092 | OSSL_DEPRECATEDIN_3_0 void EVP_PKEY_meth_get_sign
      |                            ^~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:436:3: warning: 'EVP_PKEY_meth_get_decrypt' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  436 |   EVP_PKEY_meth_get_decrypt(orig_meth, &orig_pkey_rsa_decrypt_init,
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/evp.h:2120:28: note: declared here
 2120 | OSSL_DEPRECATEDIN_3_0 void EVP_PKEY_meth_get_decrypt
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:439:3: warning: 'EVP_PKEY_meth_new' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  439 |   new_meth = EVP_PKEY_meth_new(EVP_PKEY_RSA, EVP_PKEY_FLAG_AUTOARGLEN);
      |   ^~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/evp.h:1754:40: note: declared here
 1754 | OSSL_DEPRECATEDIN_3_0 EVP_PKEY_METHOD *EVP_PKEY_meth_new(int id, int flags);
      |                                        ^~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:441:3: warning: 'EVP_PKEY_meth_copy' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  441 |   EVP_PKEY_meth_copy(new_meth, orig_meth);
      |   ^~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/evp.h:1757:28: note: declared here
 1757 | OSSL_DEPRECATEDIN_3_0 void EVP_PKEY_meth_copy(EVP_PKEY_METHOD *dst,
      |                            ^~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:443:3: warning: 'EVP_PKEY_meth_set_sign' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  443 |   EVP_PKEY_meth_set_sign(new_meth, orig_pkey_rsa_sign_init, cie_pkey_rsa_sign);
      |   ^~~~~~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/evp.h:2023:28: note: declared here
 2023 | OSSL_DEPRECATEDIN_3_0 void EVP_PKEY_meth_set_sign
      |                            ^~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:444:3: warning: 'EVP_PKEY_meth_set_decrypt' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  444 |   EVP_PKEY_meth_set_decrypt(new_meth, orig_pkey_rsa_decrypt_init,
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/evp.h:2050:28: note: declared here
 2050 | OSSL_DEPRECATEDIN_3_0 void EVP_PKEY_meth_set_decrypt
      |                            ^~~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c: In function 'cie_get_rsa_method':
../sign-sdk/src/CIEEngine.c:486:5: warning: 'RSA_meth_dup' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  486 |     ops = RSA_meth_dup(RSA_get_default_method());
      |     ^~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/rsa.h:486:35: note: declared here
  486 | OSSL_DEPRECATEDIN_3_0 RSA_METHOD *RSA_meth_dup(const RSA_METHOD *meth);
      |                                   ^~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:486:5: warning: 'RSA_get_default_method' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  486 |     ops = RSA_meth_dup(RSA_get_default_method());
      |     ^~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/rsa.h:310:41: note: declared here
  310 | OSSL_DEPRECATEDIN_3_0 const RSA_METHOD *RSA_get_default_method(void);
      |                                         ^~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:488:5: warning: 'RSA_meth_set1_name' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  488 |     RSA_meth_set1_name(ops, "CIE RSA method");
      |     ^~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/rsa.h:488:27: note: declared here
  488 | OSSL_DEPRECATEDIN_3_0 int RSA_meth_set1_name(RSA_METHOD *meth,
      |                           ^~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:489:5: warning: 'RSA_meth_set_flags' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  489 |     RSA_meth_set_flags(ops, 0);
      |     ^~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/rsa.h:491:27: note: declared here
  491 | OSSL_DEPRECATEDIN_3_0 int RSA_meth_set_flags(RSA_METHOD *meth, int flags);
      |                           ^~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:490:5: warning: 'RSA_meth_set_finish' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  490 |     RSA_meth_set_finish(ops, cie_rsa_free);
      |     ^~~~~~~~~~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/rsa.h:565:5: note: declared here
  565 | int RSA_meth_set_finish(RSA_METHOD *rsa, int (*finish) (RSA *rsa));
      |     ^~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c: In function 'bind_helper':
../sign-sdk/src/CIEEngine.c:511:3: warning: 'ENGINE_set_id' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  511 |   if (!ENGINE_set_id(e, engine_cie_id) ||
      |   ^~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:495:27: note: declared here
  495 | OSSL_DEPRECATEDIN_3_0 int ENGINE_set_id(ENGINE *e, const char *id);
      |                           ^~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:512:7: warning: 'ENGINE_set_name' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  512 |       !ENGINE_set_name(e, engine_cie_name) ||
      |       ^
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:496:27: note: declared here
  496 | OSSL_DEPRECATEDIN_3_0 int ENGINE_set_name(ENGINE *e, const char *name);
      |                           ^~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:513:7: warning: 'ENGINE_set_init_function' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  513 |       !ENGINE_set_init_function(e, cie_init) ||
      |       ^
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:505:5: note: declared here
  505 | int ENGINE_set_init_function(ENGINE *e, ENGINE_GEN_INT_FUNC_PTR init_f);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:514:7: warning: 'ENGINE_set_RSA' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  514 |       !ENGINE_set_RSA(e, cie_get_rsa_method()) ||
      |       ^
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:497:27: note: declared here
  497 | OSSL_DEPRECATEDIN_3_0 int ENGINE_set_RSA(ENGINE *e, const RSA_METHOD *rsa_meth);
      |                           ^~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:515:7: warning: 'ENGINE_set_destroy_function' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  515 |       !ENGINE_set_destroy_function(e, cie_destroy) ||
      |       ^
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:503:5: note: declared here
  503 | int ENGINE_set_destroy_function(ENGINE *e,ENGINE_GEN_INT_FUNC_PTR destroy_f);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:516:7: warning: 'ENGINE_set_finish_function' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  516 |       !ENGINE_set_finish_function(e, cie_finish) ||
      |       ^
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:507:5: note: declared here
  507 | int ENGINE_set_finish_function(ENGINE *e, ENGINE_GEN_INT_FUNC_PTR finish_f);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:517:7: warning: 'ENGINE_set_pkey_meths' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  517 |       !ENGINE_set_pkey_meths(e, cie_pkey_meths) ||
      |       ^
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:522:5: note: declared here
  522 | int ENGINE_set_pkey_meths(ENGINE *e, ENGINE_PKEY_METHS_PTR f);
      |     ^~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:518:7: warning: 'ENGINE_set_cmd_defns' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  518 |       !ENGINE_set_cmd_defns(e, cie_engine_cmd_defns) ||
      |       ^
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:526:27: note: declared here
  526 | OSSL_DEPRECATEDIN_3_0 int ENGINE_set_cmd_defns(ENGINE *e,
      |                           ^~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:519:7: warning: 'ENGINE_set_ctrl_function' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  519 |       !ENGINE_set_ctrl_function(e, cie_engine_ctrl) ||
      |       ^
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:509:5: note: declared here
  509 | int ENGINE_set_ctrl_function(ENGINE *e, ENGINE_CTRL_FUNC_PTR ctrl_f);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:520:7: warning: 'ENGINE_set_load_privkey_function' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  520 |       !ENGINE_set_load_privkey_function(e, cie_load_privkey) ||
      |       ^
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:511:5: note: declared here
  511 | int ENGINE_set_load_privkey_function(ENGINE *e, ENGINE_LOAD_KEY_PTR loadpriv_f);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c:521:7: warning: 'ENGINE_set_load_pubkey_function' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  521 |       !ENGINE_set_load_pubkey_function(e, cie_load_pubkey)  //||
      |       ^
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:513:5: note: declared here
  513 | int ENGINE_set_load_pubkey_function(ENGINE *e, ENGINE_LOAD_KEY_PTR loadpub_f);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../sign-sdk/src/CIEEngine.c: In function 'engine_cie':
../sign-sdk/src/CIEEngine.c:535:3: warning: 'ENGINE_new' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  535 |   ENGINE *e = ENGINE_new();
      |   ^~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:492:31: note: declared here
  492 | OSSL_DEPRECATEDIN_3_0 ENGINE *ENGINE_new(void);
      |                               ^~~~~~~~~~
../sign-sdk/src/CIEEngine.c:538:5: warning: 'ENGINE_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  538 |     ENGINE_free(e);
      |     ^~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:493:27: note: declared here
  493 | OSSL_DEPRECATEDIN_3_0 int ENGINE_free(ENGINE *e);
      |                           ^~~~~~~~~~~
../sign-sdk/src/CIEEngine.c: In function 'engine_load_cie':
../sign-sdk/src/CIEEngine.c:548:3: warning: 'ENGINE_add' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  548 |   ENGINE_add(toadd);
      |   ^~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:328:27: note: declared here
  328 | OSSL_DEPRECATEDIN_3_0 int ENGINE_add(ENGINE *e);
      |                           ^~~~~~~~~~
../sign-sdk/src/CIEEngine.c:549:3: warning: 'ENGINE_free' is deprecated: Since OpenSSL 3.0 [-Wdeprecated-declarations]
  549 |   ENGINE_free(toadd);
      |   ^~~~~~~~~~~
/nix/store/pp2zf8bdgyz60ds8vcshk2603gcjgp72-openssl-3.3.2-dev/include/openssl/engine.h:493:27: note: declared here
  493 | OSSL_DEPRECATEDIN_3_0 int ENGINE_free(ENGINE *e);
      |                           ^~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

On ce3899414dab3297cf025bfa356dc2da426feefd

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@rnhmjoj 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ci/cie-middleware-linux/package.nix
+++ b/pkgs/by-name/ci/cie-middleware-linux/package.nix
@@ -21,13 +21,13 @@
 
 let
   pname = "cie-middleware-linux";
-  version = "1.5.2";
+  version = "1.5.6";
 
   src = fetchFromGitHub {
     owner = "M0rf30";
     repo = pname;
     rev = version;
-    sha256 = "sha256-M3Xwg3G2ZZhPRV7uhFVXQPyvuuY4zI5Z+D/Dt26KVM0=";
+    sha256 = "sha256-2P/1hQTmeQ6qE7RgAeLOZTszcLcIpa2XX1S2ahXRHcc=";
   };
 
   gradle = gradle_8;

